pkg/antctl/raw/featuregates/command.go

```diff
@@ -0,0 +1,150 @@
+	return featureGateRequest(cmd, "remote")
+}
+
+func featureGateRequest(cmd *cobra.Command, mode string) error {
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	kubeconfig.GroupVersion = &schema.GroupVersion{Group: "", Version: ""}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+	if server, err := Command.Flags().GetString("server"); err != nil {
+		kubeconfig.Host = server
+	}
+
+	k8sClientset, antreaClientset, err := raw.SetupClients(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create clientset: %w", err)
+	}
+
+	var resp []featuregates.Response
+	var client *rest.RESTClient
+
+	switch mode {
+	case runtime.ModeAgent, runtime.ModeController:
+		client, err = rest.RESTClientFor(restconfigTmpl)
+	case "remote":
+		client, err = getControllerClient(k8sClientset, antreaClientset, restconfigTmpl)
+	}
+	if err != nil {
+		return fmt.Errorf("fail to creating rest client %w", err)
+	}
+
+	if mode == runtime.ModeAgent {
+		if resp, err = getFeatureGatesRequest(client); err != nil {
+			return err
+		}
+		fmt.Println(output(resp, runtime.ModeAgent))
+	} else {
+		if resp, err = getFeatureGatesRequest(client); err != nil {
+			return err
+		}
+		var agentGates []featuregates.Response
+		var controllerGates []featuregates.Response
+		for _, v := range resp {
+			if v.Component == "agent" {
+				agentGates = append(agentGates, v)
+			} else {
+				controllerGates = append(controllerGates, v)
+			}
+		}
+		fmt.Println(output(agentGates, runtime.ModeAgent))
+		fmt.Println(output(controllerGates, runtime.ModeController))
+	}
+	return nil
+}
+
+func getControllerClient(k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, cfgTmpl *rest.Config) (*rest.RESTClient, error) {
+	controllerClientCfg, err := raw.CreateControllerClientCfg(k8sClientset, antreaClientset, cfgTmpl)
+	if err != nil {
+		return nil, fmt.Errorf("error when creating controller client config: %w", err)
+	}
+	controllerClient, err := rest.RESTClientFor(controllerClientCfg)
+	if err != nil {
+		return nil, fmt.Errorf("error when creating controller client: %w", err)
+	}
+	return controllerClient, nil
+}
+
+func getFeatureGatesRequest(client *rest.RESTClient) ([]featuregates.Response, error) {
+	var resp []featuregates.Response
+	rawResp, err := client.Get().Resource("").Name("featuregates").DoRaw(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("error when requesting feature gates list: %w", err)
+	}
+	err = json.Unmarshal(rawResp, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unmarshal feature gates list: %w", err)
+	}
+	return resp, nil
+}
+
+func output(resps []featuregates.Response, runtimeMode string) string {
+	var output strings.Builder
+	output.Write([]byte(getHeader(runtimeMode)))
+	formatter := "%-25s%-15s%-10s\n"
+	output.Write([]byte(fmt.Sprintf(formatter, "FEATUREGATE", "STATUS", "VERSION")))
```

Comment:
I think antctl has a framework to do this, why doesn't it work for this command?

Reply:
there is a framework but I didn't see it can handle raw command, and it can't check response  and split it into two parts conditionally, so I added this alone.